lgc/state/PalMetadata.cpp

```diff
@@ -447,11 +448,25 @@ void PalMetadata::finalizePipeline() {
   paClClipCntl.bits.DX_RASTERIZATION_KILL = rasterizerDiscardEnable;
   setRegister(mmPA_CL_CLIP_CNTL, paClClipCntl.u32All);
 
+  if (m_pipelineState->isGraphics()) {
+    UpdateSpiShaderColFormat();
+  }
+
   // If there are root user data nodes but none of them are used, adjust userDataLimit accordingly.
   if (m_userDataLimit->getUInt() == 0 && !m_pipelineState->getUserDataNodes().empty())
     setUserDataLimit();
 }
 
+// =====================================================================================================================
+// Get a register value in PAL metadata.  Returns 0 if the node does not have an entry.
+//
+// @param regNum : Register number
+unsigned PalMetadata::getRegister(unsigned regNum) {
+  msgpack::DocNode &node = m_registers[regNum];
```

Comment:
This will add an entry if it doesn't exist. We haven't been super consistent about this, but please use `find` instead.